### PR TITLE
#915 pass '--config=[os]' to bazel in get_all_heron_paths.sh

### PR DESCRIPTION
**Pull request by caofangkun**

### CODE REVIEW

File: website/scripts/setup.sh [hunk 1]
```diff
@@ -1,7 +1,28 @@
...
+
+PLATFORM=`platform`
+if [ $PLATFORM = darwin ]; then
+  brew update && brew install nvm && source $(brew --prefix nvm)/nvm.sh
+  nvm install node
+  curl -L https://www.npmjs.com/install.sh | sh 
+  brew list hugo || brew install hugo
+  which wget || brew install wget
+elif [ $PLATFORM = ubuntu ]; then
+  sudo apt-get install golang git mercurial python-pygments -y
```
Comment (kramasamy) [inline website/scripts/setup.sh]: @caofangkun - Wondering why you need the 'go' lang package?

File: website/scripts/setup.sh [hunk 2]
```diff
@@ -1,7 +1,28 @@
...
+  export PATH=$PATH:$GOROOT/bin:$GOPATH/bin
+  go get -u -v github.com/spf13/hugo
+elif [ $PLATFORM = centos ]; then
+  sudo yum -y install nodejs npm golang --enablerepo=epel
+  export GOROOT=/usr/lib/go
+  export GOPATH=$HOME/go
+  export PATH=$PATH:$GOROOT/bin:$GOPATH/bin
+  go get -u -v github.com/spf13/hugo
+fi
+
```
Comment (kramasamy) [inline website/scripts/setup.sh]: @caofangkun - if you install python-pygments above, why we are uninstalling and installing again?

File: website/scripts/setup.sh (hunk 1 above)
Comment (caofangkun) [inline website/scripts/setup.sh]: @kramasamy 
need ```go ``` to install ```hugo``` on Ubuntu.
```go get -u -v github.com/spf13/hugo```

File: website/scripts/setup.sh (hunk 2 above)
Comment (caofangkun) [inline website/scripts/setup.sh]: @kramasamy 
Yes , it's not necessary install ```python-pygments``` here.
fixed.

File: scripts/get_all_heron_paths.sh [hunk 3]
```diff
@@ -23,7 +23,8 @@ function query() {
 
 set +e
 # Build everything
-bazel build {heron,integration-test,tools/java}/...
+source ./detect_os_type.sh
```
Comment (billonahill) [inline scripts/get_all_heron_paths.sh]: will this work when run from both the repo root dir (i.e. `scripts/get_all_heron_paths.sh`) and from within `scripts/.`?

Comment (caofangkun) [inline scripts/get_all_heron_paths.sh]: Hi @billonahill 
relative path of ```detect_os_type.sh``` here is not indeed. 
I have fixed it. Thank you for pointing out :)

**Comment by kramasamy** [comment]
Looks good to me. @billonahill - if you are ok with it, we can merge it.

**Comment by billonahill** [comment]
Thanks @caofangkun looks good! Merging.

**Pull request merged**